cpp/src/arrow/json/test-common.h

```diff
@@ -170,7 +170,14 @@ inline static Status ParseFromString(ParseOptions options, string_view src_str,
 
 std::string PrettyPrint(string_view one_line) {
   rj::Document document;
-  document.Parse(one_line.data());
+  // RapidJSON can use SIMD operations that read past string
+  // but make ASAN complain, pad to make sure there is enough
+  // memory to keep ASAN happy (and accomodate if rapid json upgrades
+  // to SIMD instructions with 512 byte width).
+  std::string line(one_line);
+  line.reserve(line.size() + 512);
+
+  document.Parse(line.data());
```

Comment:
Factor this out?

Reply:
used different API.  Do you still think its worth factoring out?

Reply:
Not if you're not using a `reserve()` trick. Just add a comment explaining why you're careful to use that API.